### PR TITLE
[Feature] Support shallow copy for Config

### PR DESCRIPTION
**Pull request by imabackstabber**
## Motivation

since [PR#1658](https://github.com/open-mmlab/mmcv/pull/1658) support a `__deepcopy__` method for `Config`,`__copy__` method should also be implemented.

## Modification

Add `__copy__`  to `Config`

## BC-breaking (Optional)

No

## Use cases (Optional)

Shown in `test_copy` method in `tests/test_utils/test_config.py`

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcv/utils/config.py [hunk 1]
```diff
@@ -528,6 +528,15 @@ def __iter__(self):
     def __getstate__(self):
         return (self._cfg_dict, self._filename, self._text)
 
+    def __copy__(self):
+        cls = self.__class__
+        other = cls.__new__(cls)
+
+        for key, value in self.__dict__.items():
+            super(Config, other).__setattr__(key, value)
```
Comment (zhouzaida) [inline mmcv/utils/config.py:536]: Can we directly use `other.__dict__.update(self.__dict__)`?

Comment (imabackstabber) [reply mmcv/utils/config.py]: > Can we directly use `other.__dict__.update(self.__dict__)`?

i think so.Let me correct it.

**Review by imabackstabber — COMMENTED** [review]
code changed as suggested

File: tests/test_utils/test_config.py [hunk 2]
```diff
@@ -549,3 +549,16 @@ def test_deepcopy():
...
+
+def test_copy():
+    cfg_file = osp.join(data_path, 'config/n.py')
+    cfg = Config.fromfile(cfg_file)
+    new_cfg = copy.copy(cfg)
+
+    assert isinstance(new_cfg, Config)
+    assert new_cfg._cfg_dict == cfg._cfg_dict
+    assert new_cfg is not cfg
+    assert new_cfg._cfg_dict is cfg._cfg_dict
```
Comment (zhouzaida) [inline tests/test_utils/test_config.py:562]:
```suggestion
    assert new_cfg is not cfg
    assert new_cfg._cfg_dict is cfg._cfg_dict
```

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**